### PR TITLE
feat: github repository module

### PR DESCRIPTION
**Pull request by tonirvega**
related to  #94

### CODE REVIEW

File: modules/github-repository/main.tf [hunk 1]
```diff
@@ -0,0 +1,41 @@
+resource "github_repository" "this" {
+  name        = var.name
+  visibility = var.visibility
+  auto_init = true
+
+  lifecycle {
+    ignore_changes = [
```
Comment (alambike) [inline modules/github-repository/main.tf:7]: I think we can remove this ignores and configure every repository imported by hand, or based in org template without the standard defaults

File: modules/github-repository/main.tf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      has_wiki,
+      id,
+      is_template,
+      merge_commit_message,
+      merge_commit_title,
+      squash_merge_commit_message,
+      squash_merge_commit_title,
+      topics,
+      vulnerability_alerts,
+      web_commit_signoff_required,
```
Comment (alambike) [inline modules/github-repository/main.tf:29]:
```suggestion
      allow_auto_merge, -> False
      description, -> Repo name
      allow_merge_commit, -> True
      allow_rebase_merge, -> True
      allow_squash_merge, -> True
      allow_update_branch, -> True
      archived, -> False
      delete_branch_on_merge, -> True
      has_discussions, -> False (a menos que el repo tenga discussions activas)
      has_downloads, -> ???
      has_issues, -> True
      has_projects, -> False
      has_wiki, -> True
      id, -> rep name
      is_template, -> False
      merge_commit_message, -> ""
      merge_commit_title, -> "feat: xxx"
      squash_merge_commit_message, -> ""
      squash_merge_commit_title, -> "feat: xxx"
      topics, ""
      vulnerability_alerts, -> False?
      web_commit_signoff_required, -> False
```

Comment (alambike) [reply modules/github-repository/main.tf:29]: Default branch => "master"

Comment (tonirvega) [reply modules/github-repository/main.tf]: i'm not sure  about this, the provider default is "main"

**Pull request closed without merging**